src/dials/algorithms/symmetry/cosym/__init__.py

```diff
@@ -382,15 +391,48 @@ def _reindexing_ops(
         coord_ids = np.arange(n_datasets * n_sym_ops)
         dataset_ids = coord_ids % n_datasets
 
-        # choose a high density point as seed
-        X = coords
-        nbrs = NearestNeighbors(
-            n_neighbors=min(11, len(X)), algorithm="brute", metric="cosine"
-        ).fit(X)
-        distances, indices = nbrs.kneighbors(X)
-        average_distance = np.array([dist[1:].mean() for dist in distances])
-        i = average_distance.argmin()
-        xis = np.array([X[i]])
+        if (
+            self.using_reference
+        ):  # if reference, choose the reference with reindexing op xyz as seed
+            xis = None
+            sel = np.where(dataset_ids == n_datasets - 1)
+            X = coords[sel]
+            for i in range(len(X)):
+                for partition in cosets.partitions:
+                    if sym_ops[i] in partition:
+                        cb_op = sgtbx.change_of_basis_op(partition[0]).new_denominators(
+                            self.cb_op_inp_min
+                        )
+                        reindexing_op = (
+                            self.cb_op_inp_min.inverse() * cb_op * self.cb_op_inp_min
+                        ).as_xyz()
+                        if reindexing_op == "x,y,z":
+                            xis = np.array([X[i]])
+                        break
+                if xis is not None:
+                    break
+            if xis is None:
+                raise ValueError("Unable to determine x,y,z mode of reference")
```

Comment:
1. checking that `self.cb_op_inp_min.inverse() * cb_op * self.cb_op_inp_min` is equal to identify op, is equivalent to simply checking that `cb_op` is the identity op
2. I think this whole block isn't actually necessary, as the first copy of the dataset should be the one that hasn't had a change of basis applied, which I think it what you are looking for here, the following should be equiavelent:
```suggestion
            xis = np.array([coords[sel][0]])
```
The order of the symmetry copies of the dataset is determined by the order of `Target.sym_ops`, and therefore the first copy should be the one with `cb_op` = `x,y,z`:
https://github.com/dials/dials/blob/2abb8d42ee323e4b82623e2d564c134693fd3b60/src/dials/algorithms/symmetry/cosym/target.py#L92